terraform/api_gateway.tf.json.template.py

```diff
@@ -173,6 +176,14 @@ def for_domain(cls, domain):
                 lambda_.name: {
                     "name": "/aws/apigateway/" + config.qualified_resource_name(lambda_.name),
                     "retention_in_days": 1827,
+                }
+            },
+            "null_resource": {
+                f'{lambda_.name}_log_group_provisioner': {
+                    "triggers": {
+                        "file_sha1": file_sha1(config.project_root + "/scripts/log_api_gateway.py"),
+                        "log_group_id": f"aws_cloudwatch_log_group.{lambda_.name}.id"
```

Comment:
AFAICT, this is just a fixed string, not the actual id. And it's not a reference to another resource so the dependency isn't set up. It depends on nothing so it could run at any time. Make sure you test this thoroughly.